Development capacity pool

zaibon opened the pull request:
Part of story: https://github.com/threefoldtech/home/issues/749

- [x] Review billing to make it work to pay capacity pool
- [x] Figure out how we can still allow client to work with reservation concept (list of workloads)
- [x] refactor network primitive
- [ ] [refactor chat flow once client and SAL is updated](https://github.com/threefoldtech/jumpscaleX_libs/pull/243)

zaibon committed:
explorer client: add new types to support capacity pools

zaibon committed:
zos SAL: update sal to support new capacity pool flow

- add some method to manage capacity pool
- add methods to deploy worklaods linked to a pool
- remove reservation related method that are not longer allowed
- implement new signature system

grimpy commented on JumpscaleLibs/clients/explorer/pools.py:
```diff
@@ -0,0 +1,38 @@
+
+    def list(self, customer_tid=None, next_action=None, page=None):
+        if page:
+            reservations, _ = get_page(self._session, page, self._model, self._base_url)
```
url here seems to mismatch the url in iter?

grimpy commented on JumpscaleLibs/clients/explorer/pools.py:
```diff
@@ -0,0 +1,38 @@
+        return reservations
+
+    def iter(self):
+        url = self._base_url + f"/owner/{j.me.tid}"
```
this method is used from list where customer_tid is passed while here we use j.me.tid...

grimpy commented on JumpscaleLibs/sal/zosv2/signature.py:
```diff
@@ -0,0 +1,269 @@
+    b.write(str(container.capacity.cpu))
+    b.write(str(container.capacity.memory))
+    b.write(str(container.capacity.disk_size))
+    b.write(str(container.capacity.disk_type))
```
i assume there needs to be a b.getvalue() here?

zaibon committed:
explorer client: finish to update the code to support new explorer
endpoints

zaibon committed:
zos SAL: implement capacity pool extension

zaibon committed:
zos SAL: add docstring to pools module

zaibon committed:
zos SAL: re-introduce the concept of reservation

Now the reservation is just a list of workloads.
It allows to keep the API of the SAL pretty much untouched

zaibon committed:
add capacity_pool_id argument to all workload creation methods

zaibon committed:
zos SAL: update payout_farmers method to use pool billing object

zaibon committed:
add missing return to container_challenge function

zaibon committed:
zos SAL: fix memo text send when paying for pool capacity

zaibon committed:
zos SAL: delete id module

zaibon committed:
clean up trailing white space

zaibon committed:
remove unsed import

zaibon committed:
add workloadInfo.reference into the workload signature challenge
generation

zaibon committed:
zos SAL: implement signing delete request to decomission workload

zaibon committed:
remove expiration_provisioning field from workload info

zaibon committed:
sal SOZ: fix filtering based on workload next_action value

zaibon committed:
more remove of expiration_provisioning field

zaibon committed:
fix signature generation for container workload

zaibon committed:
more fixes in workloads signature generation

zaibon committed:
fix signature creation for network resources

zaibon committed:
also add gateways into the list of nodes IDs when creating a pool

zaibon committed:
zos SAL: add module to convert old style reservation to new style workload

zaibon committed:
zos SAL: handle already converted workloads
